fcrepo-clients/src/main/java/edu/unc/lib/dl/util/FilePremisLogger.java

```diff
@@ -0,0 +1,181 @@
+	
+	private File premisFile;
+	private PID objectPid;
+	private Model model;
+
+	private Repository repository;
+
+	public FilePremisLogger(PID pid) {
+		this.objectPid = pid;
+	}
+
+	public FilePremisLogger(PID pid, File file, Repository repository) {
+		this.objectPid = pid;
+		this.premisFile = file;
+		this.repository = repository;
+	}
+
+	/**
+	 * Allows for an arbitrary timestamp to be set for a premis event
+	 * 
+	 * @param eventType
+	 * @return PremisEventBuilder
+	 */
+	@Override
+	public PremisEventBuilder buildEvent(Resource eventType, Date date) {
+		if (date == null) {
+			date = new Date();
+		}
+
+		return new PremisEventBuilder(repository.mintPremisEventUrl(objectPid),
+				eventType, date, this);
+	}
+
+	/**
+	 * Returns an instance of buildEvent with the timestamp automatically set to the current time
+	 * 
+	 * @param eventType
+	 * @return PremisEventBuilder
+	 */
+	@Override
+	public PremisEventBuilder buildEvent(Resource eventType) {
+		return new PremisEventBuilder(repository.mintPremisEventUrl(objectPid),
+				eventType, new Date(), this);
+	}
+
+	/**
+	 * Adds an event to the log file
+	 * 
+	 * @param eventResc
+	 * @return
+	 */
+	@Override
+	public PremisLogger writeEvent(Resource eventResc) {
+		// Add the event to the model for this event log
+		Model model = getModel().add(eventResc.getModel());
+
+		if (premisFile != null) {
+			// Persist the log to file
+			try (FileOutputStream rdfFile = new FileOutputStream(premisFile)) {
+				RDFDataMgr.write(rdfFile, model, RDFFormat.TURTLE_PRETTY);
+			} catch (IOException e) {
+				throw new ObjectPersistenceException("Failed to stream PREMIS log to file for " + objectPid, e);
+			}
+		}
+
+		return this;
+	}
+
+	/**
+	 * Returns the Model containing events from this logger
+	 * 
+	 * @return
+	 */
+	public Model getModel() {
+		if (model != null) {
+			return model;
+		}
+
+		Model model = ModelFactory.createDefaultModel();
+
+		if (premisFile != null && premisFile.exists()) {
+			InputStream in = FileManager.get().open(premisFile.getAbsolutePath());
+			model.read(in, null, "TURTLE");
+		}
+
+		return model;
+	}
+
+	@Override
+	public List<PID> listEvents() {
+		List<PID> eventPids = new ArrayList<>();
+
+		// Find all of the individual events and turn their identifiers into pids
+		ResIterator eventIt = model.listResourcesWithProperty(Premis.hasEventType);
+		while (eventIt.hasNext()) {
+			Resource eventResc = eventIt.nextResource();
+			PID eventPid = PIDs.get(eventResc.getURI());
+
+			eventPids.add(eventPid);
+		}
+
+		return eventPids;
+	}
+
+	@Override
+	public List<PremisEventObject> getEvents() {
+		List<PremisEventObject> events = new ArrayList<>();
+
+		// Find all of the events and construct a list of PremisEventObjects from them.
+		ResIterator eventIt = model.listResourcesWithProperty(Premis.hasEventType);
+		while (eventIt.hasNext()) {
+			Resource eventResc = eventIt.nextResource();
```

Comment:
I may be reading this incorrectly, but won't the code as written lead to the first Resource being ignored? In other words, wouldn't we want to get the eventPid from the Resource eventIt is currently pointing to when it's first created, and then call nextResource() afterwards?

ResIterator eventIt = model.listResourcesWithProperty(Premis.hasEventType);
Resource eventResc = eventIt.getResource(); // or whatever the appropriate method is for this action
 +		while (eventIt.hasNext()) {
 +		      PID eventPid = PIDs.get(eventResc.getURI());	
                      eventResc = eventIt.nextResource();